packages/kbn-discover-contextual-components/src/data_types/logs/components/cell_actions_popover.tsx

```diff
@@ -158,13 +162,21 @@ export interface FieldBadgeWithActionsProps
   icon?: EuiBadgeProps['iconType'];
 }
 
+interface FieldBadgeWithActionsDependencies {
+  core?: CoreStart;
+  share?: SharePluginStart;
```

Comment:
Maybe my confusion about binding the dependencies early below is driven by the fact that I can't seem to figure out why we need those dependencies to be added here and all upstream usage sites. I'm surely missing something obvious, am I not?

Reply:
We need `share` and `core` for the [`ServiceNameBadgeWithActions` component](https://github.com/elastic/kibana/blob/main/src/plugins/discover/public/components/data_types/logs/service_name_badge_with_actions.tsx#L21). Previously this was accessed via context directly (as it lived in Discover). I changed this to accept the dependencies as props as everything was moved to the new package and we can no longer rely on that context.

The `SummaryCellPopover` ultimately uses this for the component map creation [from this line now](https://github.com/elastic/kibana/blob/main/src/plugins/discover/public/components/data_types/logs/summary_column/summary_column.tsx#L69).